src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -360,7 +361,7 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                         logsEnumerable = logsEnumerable.Concat(_request.AdditionalMessages);
                     }
                     var logs = logsEnumerable
-                        .ToList();
+                        .AsList();
```

Comment:
This change is unnecessary. There's no functional change here. 
LINQ generates enumerates which had to be converted to List.